api/views/maps.py

```diff
@@ -0,0 +1,77 @@
+from api import app
+from flask import Blueprint, request
+from api import db
+from api.models import Map
+import json
+from flask import jsonify
+#from api.utils import InvalidUsage
+from sqlalchemy import func
+# import time
+# from datetime import date
+# import uuid
+from flask_sqlalchemy import SQLAlchemy
+from api.utils import create_response
+
+
+
+mod = Blueprint('maps', __name__)
+
+
+#none of these have been comprehsnsively tested yet
+
+@app.route('/maps', methods = ['GET'])
+# Get all map years.
+# get function is definitely not functional
+def get_map_years():
+    if request.method == 'GET':
+        data_in = to_dict(Map.query.all())
+        year_list = []
+        for k in data_in.keys():
+            year_list.append(data_in[k].map_year)
+        return create_response(data = year_list)
+
+
+@app.route('/maps', methods = ['POST'])
+# Post a Map
+def create_map():
+    if request.method == 'POST':
+        data = request.get_json()
+        if data is None:
+            return create_response(status = 400, message = 'failed to enter any data')
+# do I have to worry any (None) data
+        map_year = data.get('map_year')
+        image_url = data.get('image_url')
+        if image_url is None:
+            return create_response(status = 400, message = 'no image_url entered')
+        if map_year is None:
+            return create_response(status = 400, message = 'no map_year entered')
+        # original code implementation -> data_in = Maps(
+        #    image_url = (json_dict['image_url']),
+        #    map_year = (int)(json_dict['year'])
+        #)
+        data_in = Map(
```

Comment:
use the [`row_constructor`](https://github.com/hack4impact-uiuc/nnb-backend-v2/blob/master/api/utils.py#L27) util function to create new instances. this reduces the redundancy of needing to send in each param by its name.

see an example of it being used [here](https://github.com/hack4impact-uiuc/nnb-backend-v2/blob/master/api/views/pois.py#L92)

Reply:
also, i think you could use a more descriptive name instead of `data_in`, like `created_map`